File: cpp/sprint5/F/code.cpp
#ifndef REMOTE_JUDGE
struct Node {  
  int value;  
  const Node* left = nullptr;  
  const Node* right = nullptr;
  Node(int value, Node* left, Node* right) : value(value), left(left), right(right) {}
};  
#endif

#ifdef REMOTE_JUDGE
#include "solution_tree.h"
#endif
#include <cmath>
#include <iostream>
#include <cassert>

using namespace std;

int Solution(const Node* root) {
    // Your code
    // “ヽ(´▽｀)ノ”
} 

#ifndef REMOTE_JUDGE
void test() {
    Node node1({1, nullptr, nullptr});
    Node node2({4, nullptr, nullptr});
    Node node3({3, &node1, &node2});
    Node node4({8, nullptr, nullptr});
    Node node5({5, &node3, &node4});
    assert(Solution(&node5) == 3);
}


int main() {
  test();
}
#endif

File: go/sprint5/F/code.go
package main

// <template>
type Node struct {
	value  int
	left   *Node
	right  *Node
}
// <template>

func Solution(root *Node) int {
	// Your code
	// “ヽ(´▽｀)ノ”
}

func test() {
	node1 := Node{1, nil, nil}
	node2 := Node{4, nil, nil}
	node3 := Node{3, &node1, &node2}
	node4 := Node{8, nil, nil}
	node5 := Node{5, &node3, &node4}
	if Solution(&node5) != 3 {
		panic("WA")
	}
}
